### PR TITLE
Add flashtext entity extractor

### PR DESCRIPTION
**Pull request by koernerfelicia**
Closes https://github.com/RasaHQ/rasa-nlu-examples/issues/69

Still TODO:
- [ ] benchmark this vs Regex
- [x] docs

### CODE REVIEW

**Comment by koernerfelicia** [comment]
I decided not to extend this from `RegexEntityExtractor` - for one I don't think we have the option of adding the parameter `use_word_boundaries` to FlashText, and the `patterns` parameter for `Regex` doesn't map to the `lookups` for `FlashText`. As always I'm open to suggestions, though.

File: tests/test_extractors/test_flashtext_entity_extractor.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
+from typing import Text, List, Dict, Any
+
```
Comment (koaning) [inline tests/test_extractors/test_flashtext_entity_extractor.py:2]: Could you add a smoketest? I've got some Rasa pipelines with the component in there. The goal of these tests is to run the most basic end-to-end check by running a specific `config.yml` file that contains the component. It's not meant as a well-understood test, but if something is wrong then it is a useful signal for further investigation.

File: rasa_nlu_examples/extractors/flashtext_entity_extractor.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    ENTITY_ATTRIBUTE_END,
+    ENTITY_ATTRIBUTE_VALUE,
+    ENTITIES,
+)
+from rasa.shared.nlu.training_data.message import Message
+from rasa.shared.nlu.training_data.training_data import TrainingData
+
+from flashtext import KeywordProcessor
+
+
```
Comment (koaning) [inline rasa_nlu_examples/extractors/flashtext_entity_extractor.py:25]: Might be good to add documentation as well with an example config.yml. We're using `mkdocs` for this. You can run the docs locally via `mkdocs serve`.

**Review by koaning — CHANGES_REQUESTED** [review]
Thanks! 

On the face of it, this looks great! Just a few minor things to tick off. 

- [x] Could you add docs? Might be good to add documentation as well with an example config.yml. We're using `mkdocs` for this. You can run the docs locally via `mkdocs serve`. 
- [x] Could you add a smoke test? 
- [ ] If you're interested in doing a benchmark, it might be good to make an example with the name lists in this package. The largest name list is the one from the [US](https://github.com/RasaHQ/rasa-nlu-examples/tree/main/data/namelists/usa). 
- [x] I think there are some style checks that aren't passing.

File: rasa_nlu_examples/extractors/flashtext_entity_extractor.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
+import os
+from typing import List, Type, Optional, Dict, Text, Any
+
+import rasa.shared.utils.io
+
+from rasa.nlu.components import Component
+from rasa.nlu.config import RasaNLUModelConfig
+from rasa.nlu.extractors.extractor import EntityExtractor
+from rasa.nlu.extractors.regex_entity_extractor import RegexEntityExtractor
```
Comment (koaning) [inline rasa_nlu_examples/extractors/flashtext_entity_extractor.py:9]: Are we using this internally?

Comment (koernerfelicia) [reply rasa_nlu_examples/extractors/flashtext_entity_extractor.py:9]: Which part do you mean? The first two are used for typing, the third is extended by `FlashTextEE`, and the last I don't use, so should remove :)

Comment (koaning) [reply rasa_nlu_examples/extractors/flashtext_entity_extractor.py:9]:
```
from rasa.nlu.extractors.regex_entity_extractor import RegexEntityExtractor
```

The `RegexEntityExtractor` isn't used inside of this `FlashTextEntityExtractor`, or am I mistaken?

Comment (koaning) [reply rasa_nlu_examples/extractors/flashtext_entity_extractor.py:9]: Yes, please :)

**Review by koaning — APPROVED** [review]

**Comment by koaning** [comment]
LGTM! Thanks! Let me know once you've got a repo with benchmarks, that'd make for a cool algorithm whiteboard video!

**Pull request merged**